app/lib/bk/compat/parsers/bitrise/github_status.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module BK
+  module Compat
+    module BitriseSteps
+      # Handles the translation of github-status
+      class Translator
+        def translate_github_status(inputs)
+          raise 'Invalid github-status step configuration!' unless required_github_status_inputs(inputs)
+
+          params = extract_github_status_params(inputs)
+          {
+            'notify' => generate_github_status_notify(params),
+            'commands' => generate_github_status_command(params)
+          }
+        end
+
+        private
+
+        def extract_github_status_params(inputs)
+          {
+            context: inputs['context'] || 'default',
+            description: inputs['description'] || '',
+            status: determine_status(
+              inputs['set_specific_status'],
+              inputs['build_status'],
+              inputs['pipeline_build_status']
+            ),
+            target_url: inputs['pipeline_build_url'] || inputs['build_url'] || '${BUILDKITE_BUILD_URL}'
+          }
+        end
+
+        def generate_github_status_notify(params)
+          [
+            {
+              'github_commit_status' => {
+                'context' => params[:context]
+              }
+            }
+          ]
+        end
+
+        # Command step to update commit status correctly
+        def generate_github_status_command(params)
+          [
+            "buildkite-agent meta-data set github_commit_status_description '#{params[:description]}'",
+            "buildkite-agent meta-data set github_commit_status_status '#{params[:status]}'",
+            "buildkite-agent meta-data set github_commit_status_target_url '#{params[:target_url]}'"
+          ]
+        end
```

Comment:
I don't think that these provide any kind of value to the translation. What's more, meta-data is per-build not per step so they are very likely to be overwritten over and over again for no one's benefit.